apps/workspace-host/src/interface.ts

```diff
@@ -859,64 +905,67 @@ async function initSequence(workspace_id, useInitialZip, res) {
     await sqldb.queryOneRowAsync(sql.select_workspace, { workspace_id })
   ).rows[0];
 
-  const workspace = {
-    id: workspace_id,
-    course_id,
-    institution_id,
-    version,
-    launch_uuid: uuid,
-    local_name: `workspace-${uuid}`,
-    remote_name: `workspace-${workspace_id}-${version}`,
-  };
-
   logger.info(`Launching workspace-${workspace_id}-${version} (useInitialZip=${useInitialZip})`);
   try {
     // Only errors at this level will set host to unhealthy.
 
-    try {
-      debug('init: configuring workspace');
-      workspace.launch_port = await _allocateContainerPort(workspace);
-      workspace.settings = await _getWorkspaceSettings(workspace.id);
-    } catch (err) {
-      logger.error(`Error configuring workspace ${workspace_id}`, err);
-      safeUpdateWorkspaceState(
-        workspace_id,
-        'stopped',
-        'Error configuring workspace. Click "Reboot" to try again.',
-      );
-      return; // don't set host to unhealthy
-    }
+    const workspace = await run(async () => {
+      try {
+        debug('init: configuring workspace');
+        return {
+          id: workspace_id,
+          course_id,
+          institution_id,
+          version,
+          launch_uuid: uuid,
+          local_name: `workspace-${uuid}`,
+          remote_name: `workspace-${workspace_id}-${version}`,
+          launch_port: await _allocateContainerPort(workspace_id),
+          settings: await _getWorkspaceSettings(workspace_id),
```

Comment:
Nice rewrite. When I had initially looked at this, I thought it would require more state than this, but I must have been mistaken.